### ISSUE TITLE
Filtering happens after event LIMIT ? 

### ISSUE THREAD

**Issue by johannagnarsson** [sensu/grafana-sensu-go-datasource#12]
If this behavior is working as intended then feel free to close this, but I would imagine the LIMIT  would apply after any filtering occurs.
 
In my current sensu-go setup (5.14.1) I have around ~14k events at any given moment according to the web UI.

I have the following filters in place:
![image](https://user-images.githubusercontent.com/15679446/67589887-6e154000-f70e-11e9-8cd1-69cfd45e6fa9.png)

In the Sensu web UI, if I select 'Status:Incident' as the filter, I get 3 events.
![Screenshot from 2019-10-25 10-04-34](https://user-images.githubusercontent.com/15679446/67590182-2e9b2380-f70f-11e9-8147-05a08c8cdf97.png)

In Grafana If I lower the LIMIT value to below the total number of events, I will get 0 events returned.
![image](https://user-images.githubusercontent.com/15679446/67590296-89347f80-f70f-11e9-8505-82065deb12f7.png)

Bring the LIMIT to above the number of events and I get the 3 incidents listed.

![Screenshot from 2019-10-25 10-10-45](https://user-images.githubusercontent.com/15679446/67590403-d0227500-f70f-11e9-8925-a6076ca62ad4.png)

This possibly has to do with the fact that the checks start with Z and would probably be at the end of that list.

Using sensugo-datasource version 1.0.2 and grafana 6.4.3
```
# grafana-cli plugins ls
installed plugins:
sensu-sensugo-datasource @ 1.0.2 
# grafana-cli -v
Grafana cli version 6.4.3
```

**Comment by blufor** [comment]
I can verify the same behavior with the exact versions

**Comment by mariusoe** [comment]
Hi,

yes, this behavior is correct.
It is like this because the limit defines the number of elements which are fetched by the data source from the Sensu Go backend. The filtering is then applied by the data source itself (in the browser). This has been done because at the time of implementing this feature, it was the only way of achieve such functionallity. This means, the data source have to fetch all data to be able to correctly filter and aggregating data.

**Comment by Alexc0007** [comment]
Hi, i have the same issue.
any way to get around it? to fetch all data

**Comment by mariusoe** [comment]
The problem of missing data due to client-side filtering can be solved starting with version 1.1.0 of the data source by using server-side filtering (response filtering). See also #30 

Closed by #36
